pkg/services/authn/authnimpl/sync/oauth_token_sync.go

```diff
@@ -113,15 +125,47 @@ func (s *OAuthTokenSync) SyncOauthTokenHook(ctx context.Context, identity *authn
 
 const maxOAuthTokenCacheTTL = 10 * time.Minute
 
-func getOAuthTokenCacheTTL(t time.Time) time.Duration {
-	if t.IsZero() {
+func getOAuthTokenCacheTTL(accessTokenExpiry, idTokenExpiry time.Time) time.Duration {
```

Comment:
Thanks for the change, I skimmed through real quck and I think this function doesn't return the correct TTL when the `accessTokenExpiry` is zero (never expires) and there is an ID token provided (when there is an ID Token provided the expiry always set, because it is required by the spec). I quickly added a table of the expected return values from this function based on the inputs. Could you please add tests for the different scenarios and review my table below?

| Access Token Expiration | ID Token Expiration | Result |
| ------------- | ------------- |----|
| 0  | 0  | maxOAuthTokenCacheTTL (this is the case when no ID token provided) |
| 0  | Jul 5  | min(idTokenTTL, maxOAuthTokenCacheTTL) |
| Jul 5 | 0 | min(accessTokenTTL, maxOAuthTokenCacheTTL) (no ID token) |
| Jul 5 | Jul 4 | min(min(accessTokenTTL, idTokenTTL), maxOAuthTokenCacheTTL) |